ada_2_fhir-r4/cio/2.0.0/payload/cio-ReplyProposalContraIndication.xsl

```diff
@@ -0,0 +1,138 @@
+See the GNU Lesser General Public License for more details.
+
+The full text of the license is available at http://www.gnu.org/copyleft/lesser.html
+-->
+<xsl:stylesheet exclude-result-prefixes="#all" xmlns="http://hl7.org/fhir"
+    xmlns:util="urn:hl7:utilities" xmlns:f="http://hl7.org/fhir"
+    xmlns:xd="http://www.oxygenxml.com/ns/doc/xsl" xmlns:nf="http://www.nictiz.nl/functions"
+    xmlns:nm="http://www.nictiz.nl/mappings" xmlns:uuid="http://www.uuid.org"
+    xmlns:xs="http://www.w3.org/2001/XMLSchema" xmlns:xsl="http://www.w3.org/1999/XSL/Transform"
+    version="2.0">
+
+    <xsl:variable name="profileNameCioReplyProposalContraIndication"
+        >cio-ReplyProposalContraIndication</xsl:variable>
+
+    <xd:doc>
+        <xd:desc>Create a cio-ReplyProposalContraIndication instance as a Communication FHIR
+            instance from the ada element antwoord_voorstel_contra_indicatie.</xd:desc>
+        <xd:param name="in">ADA element as input. Defaults to self.</xd:param>
+        <xd:param name="subject">Optional ADA instance or ADA reference element for the
+            patient.</xd:param>
+    </xd:doc>
+    <xsl:template name="cio-ReplyProposalContraIndication" as="element(f:Communication)?">
+        <xsl:param name="in" select="." as="element()?"/>
+        <xsl:param name="subject" select="../../patient" as="element()?"/>
+
+        <xsl:for-each select="$in">
+            <Communication>
+                <xsl:variable name="registrationInformation"
+                    select="../../bouwstenen/registratie_informatie[@id = current()/registratie_informatie/@value]"/>
+                <xsl:variable name="identificationNumber"
+                    select="$registrationInformation/identificatienummer"/>
+                <xsl:variable name="author" select="$registrationInformation/auteur/*"/>
+                <xsl:variable name="creationDateTime"
+                    select="$registrationInformation/ontstaans_datum_tijd"/>
+
+                <xsl:call-template name="insertLogicalId">
+                    <xsl:with-param name="profile"
+                        select="$profileNameCioReplyProposalContraIndication"/>
+                </xsl:call-template>
+                <meta>
+                    <profile
+                        value="{concat($urlBaseNictizProfile, $profileNameCioReplyProposalContraIndication)}"
+                    />
+                </meta>
+
+                <xsl:for-each select="$identificationNumber[@value]">
+                    <identifier>
+                        <xsl:call-template name="id-to-Identifier">
+                            <xsl:with-param name="in" select="."/>
+                        </xsl:call-template>
+                    </identifier>
+                </xsl:for-each>
+
+                <xsl:for-each select="relatie_voorstel_contra_indicatie/identificatie[@value | @root | @nullFlavor]">
+                    <basedOn>
+                        <valueReference>
```

Comment:
The element `.basedOn` has data type Reference, hence it does not contain a `.valueReference` element as child. Such an element is only used within e.g. an extension or another polymorphic `.value[x]` element that can have the Reference data type.